test/compiler.ts

```diff
@@ -751,7 +751,7 @@ function runTests (solc, versionText) {
       t.test('compiling standard JSON (invalid JSON)', function (st) {
         const output = JSON.parse(solc.compile('{invalid'));
         // TODO: change wrapper to output matching error
-        st.ok(expectError(output, 'JSONError', 'Line 1, Column 2\n  Missing \'}\' or object member name') || expectError(output, 'JSONError', 'Invalid JSON supplied:'));
+        st.ok(expectError(output, 'JSONError', '[json.exception.parse_error.101] parse error at line 1, column 2: syntax error while parsing object key - invalid literal; last read: \'{i\'; expected string literal'));
```

Comment:
I don't know. I don't see any value in removing such information.

Reply:
We discussed this on the chat, so I'll just repost the relevant bits here:

> It would be better not to have this if it's easy to get rid of. It's something that would be more relevant in a debug message, but this is not an internal error, it's a normal parsing error meant for the end user. Those should be formatted without extra cruft, with things like error type being properly accounted for via the fields we already have.

> I mean, I was assuming it must be pretty simple to do. Like, instead of printing `exception.what()` print just the field containing the message.
> But actually not sure now if that message is even exposed. https://json.nlohmann.me/api/basic_json/parse_error/ does not mention it.
> Also, apparently this is less of an internal detail than I expected. The numeric codes are even officially documented: https://json.nlohmann.me/home/exceptions/
>
> So maybe it's ok after all. Just the way it's formatted makes it look like something internal. The class name there is just redundant. The error codes seem to be unique and showing that would have been enough.
>
> Also it bothers me a bit how this introduces a new layer of error codes that's not connected to our own error codes.
I don't think the compiler should really be leaking internal implementation like that in its errors.
> On the other hand doing it properly might not be easy, so I'm not saying to jump into it. We probably don't have time for that. Just still, I don't like it :)